### PR TITLE
Fix defensive-contribution scoring for custom-vs-FPL position mismatches

### PR DESCRIPTION
**Pull request by timmat118**
### The bug
Kammy scored defensive contribution off FPL's aggregated `defensive_contribution` field. But FPL computes that number using **its own** position: recoveries are **included** for FPL midfielders/forwards and **excluded** for FPL defenders. Kammy then applied a threshold chosen by **our custom** position — so whenever the two position systems disagree, the recoveries end up on the wrong side of the calculation.

Verified against live FPL data — the field is CBI+tackles for FPL defenders and CBI+tackles+**recoveries** for FPL mids/fwds:
```
DEF Gabriel  DC 277 = CBI+TK (no recoveries)
MID Saka     DC 184 = CBI+TK+REC  (68 without recoveries)
```

### Impact (e.g. Matheus Nunes — mid by us, defender by FPL)
- FPL's field excludes recoveries; our **MID** rule expects CBIRT (with recoveries) vs a threshold of 12 → he was **under-credited** and missed points he should earn.
- The mirror case (our cb/fb, FPL mid) **over-credited**.

### The fix
The raw components (`clearances_blocks_interceptions`, `tackles`, `recoveries`) are already in the FPL API responses we fetch — we just weren't mapping them. This PR:
1. Carries those raw fields through `FplPlayerGameweekData` → `PlayerGameweekStatsData` / `GameweekStatWithPoints` and the data converters.
2. Computes **CBIT** (cb/fb) or **CBIRT** (mid, + recoveries) in `calculateDefensiveContribution`, keyed off **our custom position** — no longer trusting FPL's aggregate.

This mirrors the reference (Lovable) calculation and flows through the `player-gw-points` sheet, team/league scoring, and the cup.

### Notes
- No new typecheck errors (also fixes one pre-existing missing-field error in a `createEmptyStats` factory).
- 89 tests pass, including new coverage for **both** position-mismatch directions and the "recoveries excluded for defenders / included for mids" rule.
- Regenerating `player-gw-points` (admin) applies the corrected values to existing gameweeks.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by timmat118** [comment]
### Worked example: Matheus Nunes (+30 points)

A concrete case of the mismatch this fixes. Nunes is **`DEF` in FPL** but **`mid` in our Players sheet**.

His FPL `defensive_contribution` field (what the old code read) is CBI+tackles **without recoveries**, because FPL treats him as a defender. Against our **MID** threshold of 12 that almost never qualified. Computing CBIRT ourselves (CBI+tackles+**recoveries**) for his custom position, he clears 12 nearly every week:

```
GW | CBI TK REC | FPL_DC | OLD | NEW
 8 |  3   3   8 |    6   |  0  |  2
 9 |  5   2   5 |    7   |  0  |  2
11 |  9   2   4 |   11   |  0  |  2
16 |  6   2   4 |    8   |  0  |  2
19 |  6   5  12 |   11   |  0  |  2
 … 15 qualifying gameweeks total
SEASON DefCon points:  OLD = 0   NEW = 30   DELTA = +30
```

Under the old logic he earned **0** defensive-contribution points all season; with the fix he earns **30**. Values pulled live from `element-summary/407` — note `FPL_DC` equals CBI+tackles exactly (recoveries excluded), confirming FPL bakes the metric to its own position.

File: draft/app/scoring/lib/calculations.test.ts [hunk 1]
```diff
@@ -1,5 +1,56 @@
...
+        expect(calculateDefensiveContribution(raw(6, 4, 9), 'fb')).toBe(1); // recoveries irrelevant
+    });
+
+    it('does NOT count recoveries for a defender (CBIT below 10 stays 0 despite recoveries)', () => {
+        // 8 CBIT + 5 recoveries — recoveries must be ignored for defenders, so still < 10
+        expect(calculateDefensiveContribution(raw(4, 4, 5), 'cb')).toBe(0);
+    });
+
+    it('awards a midfielder +2 when CBIRT reaches 12, including recoveries', () => {
+        expect(calculateDefensiveContribution(raw(5, 3, 4), 'mid')).toBe(2); // 8 CBIT + 4 rec = 12
```
Comment (peter-mouland) [inline draft/app/scoring/lib/calculations.test.ts]: what does mid do when it reaches 10?

i think we're missing the 'counter' + 'edge-case' tests which proves these things dont happen in the oether categories, or any earlier than stated

e.g.
here i'd expect
        expect(calculateDefensiveContribution(raw(5, 1, 4), 'mid')).toBe(0); 


and hen above i'f expect (in 'awards a defender (cb/fb) +1 when CBIT reaches 10, excluding recoveries')
        expect(calculateDefensiveContribution(raw(6, 4, 0), 'mid')).toBe(1); 

so a lot more eassertions.

File: draft/app/scoring/lib/calculations.test.ts [hunk 2]
```diff
@@ -1,5 +1,56 @@
...
+const raw = (clearancesBlocksInterceptions: number, tackles: number, recoveries: number) => ({
+    clearancesBlocksInterceptions,
+    tackles,
+    recoveries,
+});
+
+describe('calculateDefensiveContribution', () => {
+    it('awards a defender (cb/fb) +1 when CBIT reaches 10, excluding recoveries', () => {
+        expect(calculateDefensiveContribution(raw(6, 4, 0), 'cb')).toBe(1); // 10 CBIT
+        expect(calculateDefensiveContribution(raw(6, 4, 9), 'fb')).toBe(1); // recoveries irrelevant
```
Comment (peter-mouland) [inline draft/app/scoring/lib/calculations.test.ts]: I'd probably only want to see 1 input changed at a time, recoveries or position.

you could keep this in here, but then add the missing assertion, something like
        expect(calculateDefensiveContribution(raw(6, 4, 0), 'fb')).toBe(1);

this makes it clear what's impacting, or not, the output

File: draft/app/scoring/lib/calculations.ts [hunk 3]
```diff
@@ -20,14 +20,28 @@ const baselineStats: Points = {
     total: 0,
 };
 
+// Positions that count recoveries towards their defensive-contribution total (CBIRT),
+// mirroring FPL's midfielder/forward rule. Defenders (cb/fb) use CBIT only.
+const RECOVERIES_COUNT_FOR: CustomPosition[] = ['mid', 'wa', 'ca'];
```
Comment (peter-mouland) [inline draft/app/scoring/lib/calculations.ts]: wa and ca are included here but neither has a defensiveContribution rule in rules.ts, so these are currently unreachable (the early guard returns 0 first).

If that's intentional future-proofing for if those positions ever get a rule, worth a comment saying so, otherwise `mid` alone would be less confusing.

what do you think?

File: draft/app/scoring/lib/calculations.test.ts (hunk 1 above)
Comment (timmat118) [reply draft/app/scoring/lib/calculations.test.ts]: Added boundary and counter cases in the latest commit. One correction on the example though: for `mid`, `raw(6,4,0)` is CBIRT 10 against a threshold of 12, so it returns 0, not 1 (the `fb` version is 1 because fb's threshold is 10). The new tests make that explicit: mid 11 -> 0, mid 12 -> 2; fb 9 -> 0, fb 10 -> 1.

File: draft/app/scoring/lib/calculations.test.ts (hunk 2 above)
Comment (timmat118) [reply draft/app/scoring/lib/calculations.test.ts]: Done. Each case now changes a single input at a time: one holds the raw stats and varies only position, one varies only CBIT (recoveries fixed at 0), one varies only recoveries for a mid. Added the `raw(6,4,0),'fb') -> 1` assertion as part of the position-isolation test.

File: draft/app/scoring/lib/calculations.ts (hunk 3 above)
Comment (timmat118) [reply draft/app/scoring/lib/calculations.ts]: Changed to key off `mid` directly. You're right that wa/ca never reach that line (the early guard returns 0 first), so mid is the only position that counts recoveries. Left a short comment to revisit if wa/ca ever gain a rule.

**Review by peter-mouland — APPROVED** [review]
lovely
